pubsub/subscriptions/pull_concurrency.go

```diff
@@ -18,14 +18,14 @@ package subscriptions
 import (
 	"context"
 	"fmt"
-	"io"
 	"runtime"
+	"sync/atomic"
 	"time"
 
 	"cloud.google.com/go/pubsub"
 )
 
-func pullMsgsConcurrenyControl(w io.Writer, projectID, subID string) error {
+func pullMsgsConcurrenyControl(counter *int32, projectID, subID string) error {
```

Comment:
We should stick with writer as the first argument, per https://github.com/GoogleCloudPlatform/golang-samples/blob/master/CONTRIBUTING.md#print-to-an-iowriter-for-snippets

Instead, initialize a counter as a local variable and print after the end of the `Receive` call. Then, we can just test for a string that's something like "Received 5 messages".